examples/ota-requestor-app/linux/main.cpp

```diff
@@ -89,20 +155,51 @@ chip::Protocols::Id FromFullyQualified(uint32_t rawProtocolId)
     return chip::Protocols::Id(vendorId, protocolId);
 }
 
+constexpr uint16_t kOptionPeerId = 'p';
+chip::NodeId providerNodeId      = 0x0;
+
+bool HandleOptions(const char * aProgram, OptionSet * aOptions, int aIdentifier, const char * aName, const char * aValue)
+{
+    bool retval = true;
+
+    switch (aIdentifier)
+    {
+    case kOptionPeerId:
+        if (1 != sscanf(aValue, "%" PRIX64, &providerNodeId))
+        {
+            PrintArgError("%s: unable to parse Node ID: %s\n", aProgram, aValue);
+        }
+        break;
+    default:
+        PrintArgError("%s: INTERNAL ERROR: Unhandled option: %s\n", aProgram, aName);
+        retval = false;
+        break;
+    }
+
+    return (retval);
+}
+
+OptionDef cmdLineOptionsDef[] = {
+    { "peerId", chip::ArgParser::kArgumentRequired, kOptionPeerId },
+    {},
+};
+
+OptionSet cmdLineOptions = { HandleOptions, cmdLineOptionsDef, "PROGRAM OPTIONS",
+                             "  -p <peer ID>\n"
+                             "  --peerId <peer ID>\n"
```

Comment:
"peer ID" is not a thing. This should be `--providerLocation <node ID`>`. Also, we need to be clear that this works with ONLY the sample commissioning logic that assumes all examples commision in one default fabric, under one root. Otherwise, we need a way to provide the true fabric to use

Reply:
I don't mind changing this. I'll make a note in the help options about the assumptions around commissioning.

Just for my own sake - is "peer" not a valid term for CHIP? In this case I just mean "another node".

Reply:
The word for "another node" is "node", which has a "node ID"